include/aws/common/byte_buf.h

```diff
@@ -507,6 +507,8 @@ AWS_STATIC_IMPL struct aws_byte_cursor aws_byte_cursor_from_array(const void *by
  * SPECTRE mitigation purposes.
  */
 AWS_STATIC_IMPL size_t aws_nospec_mask(size_t index, size_t bound) {
```

Comment:
Can I ignore any unsigned overflow in this function? @danielsn @justinboswell @JonathanHenson @bretambrose

Reply:
Unsigned over/underflow is deliberately used as part of this function's logic. You probably should not be warning on the _presence_ of over/underflow, but it is incorrect to _assume_ that over/underflow won't occur.

Reply:
We now consider all unsigned overflow in this function to be benign, since they are used as part of the function's logic.